src/utils/tokenizers.py

```diff
@@ -53,8 +54,16 @@ def detokenize(self, tokens):
         '''
         return [self._detokenizer.unescape_xml(t) for t in tokens]
 
-
-AVAILABLE_TOKENIZERS = {
-    "OpenAI.BPE": OpenAIBPETokenizer(),
-    "MosesTokenizer": MosesTokenizer()
-}
+@functools.lru_cache(maxsize=8, typed=False)
+def get_tokenizer(tokenizer_name):
+    log.info(f"Loading Tokenizer {tokenizer_name}")
+    if tokenizer_name.startswith("bert"):
+        from pytorch_pretrained_bert import BertTokenizer
+        do_lower_case = tokenizer_name.endswith('uncased')
+        tokenizer = BertTokenizer.from_pretrained(tokenizer_name, do_lower_case=do_lower_case)
+    else:
```

Comment:
nit: flatten, just do:
```
if tokenizer_name.startswith("bert-"):
   # do stuff
elif tokenizer_name == "OpenAI.BPE":
    return OpenAIBPETokenizer()
elif tokenizer_name == "MosesTokenizer":
    return MosesTokenizer()
elif tokenizer_name == "":
   return SpaceTokenizer()  # we should implement this or wrap something, this is the right way to use ELMo
else:
   return None
```

Reply:
Also should really have MosesTokenizer require `tokenizer_name == "MosesTokenizer"` rather than falling through? But OK to leave this and put a warning if this will break anything in-flight for ACL. (If not, we should just fix it now.)

Also see note about a space tokenizer. Can leave a TODO for this, but would be good to have this function be consistent with how edge probing behaves.

Reply:
Do you mean should we exactly match MosesTokenizer as "MosesTokenizer?" I think it's better to be explicit with what type of tokenizer you want rather than defaulting on a certain tokenizer if that's what you mean.

Reply:
Yes, that's what I was thinking.